curriculum/challenges/english/20-upcoming-python/learn-data-structures-by-building-the-merge-sort-algorithm/6564aee9c077774ea49c3faf.md

```diff
@@ -0,0 +1,72 @@
+---
+id: 6564aee9c077774ea49c3faf
+title: Step 3
+challengeType: 20
+dashedName: step-3
+---
+
+# --description--
+
+The merge sort algorithm mainly performs three actions recursively: divide, conquer, and merge. Let's start with "divide".
+
+You can get the number of elements in an object using the `len()` function. You can divide the total number of elements in an object by 2 to get the mid point of the object.
+
+Here's an example:
+
+```py
+random_numbers = [2, 4, 6, 8, 10, 12, 14]
+
+numbers_mid_point = len(random_numbers) // 2
+
+print(numbers_mid_point)
+# 3
+```
+
+In the code above, we got the mid point of the `random_numbers` list by dividing its length by 2. When printed to the console, the index of the mid point is returned.
+
+First, remove the `pass` keyword.
+
+Then divide the length of `array` by `2` and assign it to the `mid_point` variable, just like in the example above.
+
+Remeber to indent your code.
+
+# --hints--
+
+There should be no `pass` keyword in the body of the function. `middle_point = len(array) // 2` should be on the same line where you removed the `pass` keyword
+
+```js
+({
+  test: () => {
+    const transformedCode = e.code.original["main.py"].replace(/\r/g, "");
+    const merge_sort = __helpers.python.getDef("\n" + transformedCode, "merge_sort");
+    const { function_body } = merge_sort;
+
+    assert.match(function_body, /middle_point\s*=\s*len\(\s*array\s*\)\s*\/\/\s*2/);
+  }
+})
+```
+
+There should be no `pass` keyword in the body of the function. `middle_point = len(array) // 2` should be on the same line where you removed the `pass` keyword
```

Comment:
Both hints are same.

Reply:
Yeah. During testing, the second test was working but I was getting the hints message for the first test. So I modified the hints to target both tests. 

This also applies to Step 12